src/embedded-checkout/iframe-event-listener.ts

```diff
@@ -77,7 +77,7 @@ export default class IframeEventListener<TEventMap extends IframeEventMap<keyof
             return;
         }
 
-        this.trigger(event.data);
+        this.trigger(event.data as any);
```

Comment:
I think you're getting an error because the `data` property of `MessageEvent` is untyped. You can get around this by specifying a type for it in the line above, i.e.:

```ts
isIframeEvent(event.data as TEventMap[keyof TEventMap], event.data.type as keyof TEventMap)
```

But in order to do this, you'll also need to fix the type in `isIframeEvent` and `IframeEvent`, i.e.:

```ts
function isIframeEvent<TEvent extends IframeEvent<TType>, TType extends string | number | symbol> {
    // ...
}
```

And
```ts
interface IframeEvent<TType extends string | number | symbol = string, TPayload = any> {
    // ...
}
```